Libraries/Text/RCTText.m

```diff
@@ -149,6 +165,101 @@ - (void)didMoveToWindow
   }
 }
 
+#pragma mark Sizing
+
+- (CGRect)updateToFitFrame:(CGRect)frame
+{
+  NSLayoutManager *layoutManager = [_textStorage.layoutManagers firstObject];
+  NSTextContainer *textContainer = [layoutManager.textContainers firstObject];
+  [textContainer setLineBreakMode:NSLineBreakByWordWrapping];
+  
+  NSRange glyphRange = NSMakeRange(0, _textStorage.length);
+
+  [self resetDrawnTextStorage];
+
+  CGSize requiredSize = [self calculateSize:_textStorage];
+  NSInteger linesRequired = [self numberOfLinesRequired:layoutManager];
+
+  __block BOOL hitMinimumScale = NO;
+  while ((requiredSize.height > CGRectGetHeight(frame) ||
+         requiredSize.width > CGRectGetWidth(frame) ||
+         (linesRequired > textContainer.maximumNumberOfLines &&
+          textContainer.maximumNumberOfLines != 0))
+         && !hitMinimumScale)
+  {
+    [_textStorage beginEditing];
+    [_textStorage enumerateAttribute:NSFontAttributeName
+                             inRange:glyphRange
+                             options:0
+                          usingBlock:^(UIFont *font, NSRange range, BOOL *stop)
+    {
+      if (font) {
+        UIFont *originalFont = [_originalString attribute:NSFontAttributeName
+                                                  atIndex:range.location
+                                           effectiveRange:&range];
+        UIFont *newFont = [font fontWithSize:font.pointSize - .5];
```

Comment:
Would need to profile this but since text layout is one of the slowest operations, my sense is that this should binary search for the appropriate size.